src/main/java/org/broadinstitute/hellbender/tools/GatherVcfs.java

```diff
@@ -345,7 +345,7 @@ private static void gatherWithBlockCopying(final List<Path> vcfs, final File out
                         boolean lastByteNewline = true;
 
                         int firstNonHeaderByteIndex = -1;
-                        while (!in.eof()) {
+                        while (!in.eof() && blockIn.available() > 0) {
```

Comment:
This history of this line is a bit unsettling. Prior to the blocked NIO PR we had `while (in.available() > 0) {`. Then with that PR it changed to `while (!in.eof()) {`, and now we have `while (!in.eof() && blockIn.available() > 0) {`. 

Are we **sure** this latest version is 100% correct? Could `in` ever reach `eof` while `blockIn` still has available bytes?